src/core/cgroup.c

```diff
@@ -1584,6 +1602,17 @@ static bool unit_get_needs_socket_bind(Unit *u) {
         return c->socket_bind_allow || c->socket_bind_deny;
 }
 
+static bool unit_get_needs_restrict_network_interfaces(Unit *u) {
+        CGroupContext *c;
+        assert(u);
+
+        c = unit_get_cgroup_context(u);
+        if (!c)
+                return false;
+
+        return c->restrict_network_interfaces != NULL;
```

Comment:
probably better: `!set_isempty(c->restrict_network_interfaces)`

(note that set_isempty() will return true if you pass NULL, so this should be totally safe)